app/services/EventReportService.scala

```diff
@@ -450,17 +452,23 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
       }
     }
 
-    for {
-      transformed1828Payload <- Future.fromTry(toTry(userAnswersJson.transform(API1828.transformToETMPData)))
-      _ <- recoverAndValidatePayload(transformed1828Payload)
-    } yield {
-      ()
+    declarationLockRepository.insertLockData(pstr, psaPspId).flatMap { isAvailable => //TODO: Rename refer to AFT
+      if (isAvailable) {
+        for {
+          transformed1828Payload <- Future.fromTry(toTry(userAnswersJson.transform(API1828.transformToETMPData)))
+          _ <- recoverAndValidatePayload(transformed1828Payload)
+        } yield {
+          NoContent
+        }
+      } else {
+        Future.successful(BadRequest)
+      }
     }
   }
 
 
-  def submitEvent20ADeclarationReport(pstr: String, userAnswersJson: JsValue, version: String)
-                                     (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[Unit] = {
+  def submitEvent20ADeclarationReport(pstr: String, psaPspId: String, userAnswersJson: JsValue, version: String)
+                                     (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[Result] = {
 
```

Comment:
As above can we change the return type to Future[Unit] for submitEvent20ADeclarationReport. Controller should be handling response codes.